src/bun.js/bindings/sqlite/JSSQLStatement.cpp

```diff
@@ -671,63 +806,174 @@ static inline bool rebindValue(JSC::JSGlobalObject* lexicalGlobalObject, sqlite3
+    //
+    // { 0: "foo", 1: "bar", "2": "baz" }
+    //
+    else if (UNLIKELY(bindings.isOnlyIndexed)) {
+        for (size_t i = 0; i < size; i++) {
+            JSValue value = target->getDirectIndex(globalObject, i);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, i + 1, "\""_s)));
+                } else {
+                    continue;
+                }
+            }
+
+            RETURN_IF_EXCEPTION(scope, JSValue());
+
+            if (!rebindValue(globalObject, db, stmt, i + 1, value, scope, clone, safeIntegers)) {
+                return JSValue();
+            }
+
+            RETURN_IF_EXCEPTION(scope, {});
+            count++;
         }
+    }
+    // Is it a simple object with no getters or setters?
+    //
+    // { foo: "bar", baz: "qux" }
+    //
+    else if (target->canUseFastGetOwnProperty(structure)) {
+        for (size_t i = 0; i < size; i++) {
+            const auto& property = bindingNames[i];
+            JSValue value = property.isEmpty() ? target->getDirectIndex(globalObject, i) : target->fastGetOwnProperty(vm, structure, bindingNames[i]);
+            if (!value && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, property.isEmpty() ? String::number(i) : property.string(), "\""_s)));
+                } else {
+                    continue;
+                }
+            }
+
+            RETURN_IF_EXCEPTION(scope, JSValue());
 
-        // Ensure this gets freed on scope clear
-        auto utf8 = WTF::String(propertyName.string()).utf8();
+            if (!rebindValue(globalObject, db, stmt, i + 1, value, scope, clone, safeIntegers)) {
+                return JSValue();
+            }
 
-        int index = sqlite3_bind_parameter_index(stmt, utf8.data());
-        if (index == 0) {
-            throwException(globalObject, scope, createError(globalObject, "Unknown parameter \"" + propertyName.string() + "\""_s));
-            return JSValue();
+            RETURN_IF_EXCEPTION(scope, {});
+            count++;
         }
+    } else {
+        for (size_t i = 0; i < size; i++) {
+            PropertySlot slot(target, PropertySlot::InternalMethodType::GetOwnProperty);
+            const auto& property = bindingNames[i];
+            bool hasProperty = property.isEmpty() ? target->methodTable()->getOwnPropertySlotByIndex(target, globalObject, i, slot) : target->methodTable()->getOwnPropertySlot(target, globalObject, property, slot);
+            if (!hasProperty && !scope.exception()) {
+                if (throwOnMissing) {
+                    throwException(globalObject, scope, createError(globalObject, makeString("Missing parameter \""_s, property.isEmpty() ? String::number(i) : property.string(), "\""_s)));
```

Comment:
and here

Reply:
Checks two lines below

https://github.com/oven-sh/bun/blob/0ba290c5395b6d9393a1069b68d4d1aae64ae259/src/bun.js/bindings/sqlite/JSSQLStatement.cpp#L936-L944